src/client/renderer/r_material.c

```diff
@@ -706,22 +712,27 @@ r_material_t *R_ConvertMaterial(cm_material_t *cm, const _Bool unref) {
 		}
 
 		R_RegisterMedia((r_media_t *) mat);
-		Cm_RefMaterial(cm);
-	}
-
-	if (unref) {
-		Cm_UnrefMaterial(cm);
 	}
 
 	return mat;
 }
 
 /**
- * @brief Loads the r_material_t from the specified texture and unreferences the cm_ it loads once.
+ * @brief Loads the r_material_t from the specified texture.
  */
 r_material_t *R_LoadMaterial(const char *name) {
 
-	return R_ConvertMaterial(Cm_LoadMaterial(name), true);
+	r_material_t *mat;
+	char material_key[MAX_QPATH];
+	StripExtension(name, material_key);
+	Cm_NormalizeMaterial(material_key, material_key, sizeof(material_key));
+	g_snprintf(material_key, sizeof(material_key), "%s_mat", material_key);
+
+	if (!(mat = (r_material_t *) R_FindMedia(material_key))) {
+		mat = R_ConvertMaterial(Cm_LoadMaterial(name), false);
+	}
+
+	return mat;
```

Comment:
But it's so elegant as it is, no extra allocations - the only thing that needs the special code is map materials since Cm loads them already, so we can just use the materials they already loaded/parsed.